cdk/src/open-data-platform/data-plane/data-import/write-violations-data.handler.ts

```diff
@@ -37,37 +42,56 @@ const COMPLIANCE_STATUS_MAP = new Map([
 
 /**
  * Inserts all rows into the violations table.
+ * @param db: Database to use
+ * @param rows: Rows to write into the db
  */
-async function insertRows(db: ConnectionPool, rows: ViolationsTableRow[]): Promise<any[]> {
-  return db.query(sql`INSERT INTO epa_violations (violation_id,
-                                                  pws_id,
-                                                  violation_code,
-                                                  compliance_status,
-                                                  start_date)
-                      VALUES ${sql.join(
-                        rows.map((row: ViolationsTableRow) => {
-                          return sql`(${row.violation_id}, ${row.pws_id},
-                                                 ${row.violation_code},
-                                                 ${row.compliance_status},
-                                                 ${row.start_date})`;
-                        }),
-                        ',',
-                      )} ON CONFLICT (violation_id) DO UPDATE;`);
+async function insertRows(db: PoolClient, rows: ViolationsTableRow[]): Promise<QueryArrayResult> {
+  const valuesToInsert: any[][] = [];
+
+  for (let row of rows) {
+    valuesToInsert.push([
+      row.violation_id,
+      row.pws_id,
+      row.violation_code,
+      row.compliance_status,
+      row.start_date,
+    ]);
+  }
+
+  // Format function below needs these to be %s (string literals) or else
+  // it produces invalid geometries.
+  const insertIntoStatement =
+    'INSERT INTO epa_violations (violation_id, pws_id,violation_code, compliance_status, start_date) ' +
+    'VALUES %L ON CONFLICT (violation_id) DO NOTHING';
+
+  try {
+    await db.query('BEGIN');
```

Comment:
Why do we need this? (same for `COMMIT` and `ROLLBACK`)

Reply:
Done